src/Header/APIHandler.h

```diff
@@ -15,7 +16,7 @@ class APIHandler
 public:
     APIHandler();
     //the start entry of all api services
-    virtual void CallingAPI(const OperateData &data);
+    virtual void CallingAPI(std::unique_ptr<OperateData> data);
```

Comment:
Note: Whenever you specify unique_ptr in the parameter, you are asking the caller to transfer the ownership of the pointer to you. If this is what you want, consider noting it in the comment because it is not obvious why CallingAPI() would need the ownership.

Reply:
As discussed in class, comment that CallingAPI will own and usually free the OperateData instance.

Reply:
Done